### PR TITLE
fix(providers): ignore URL config key vase for kv redis provider

### PR DESCRIPTION
**Pull request by vados-cosmonic**

### CODE REVIEW

**Review by joonas — APPROVED** [review]
awesome, thank you!

**Comment by vados-cosmonic** [comment]
We did it in one place, and not the other @joonas... so the idea it should be in the provider SDK is probably worth visiting.

**Comment by joonas** [comment]
@vados-cosmonic would it be reasonable to also updated the impacted examples to use a newer version (assuming we want to publish this change as a patch bump)?

File: crates/provider-keyvalue-redis/src/lib.rs [hunk 1]
```diff
@@ -301,7 +301,11 @@ impl Provider for KvRedisProvider {
             source_id, config, ..
         }: LinkConfig<'_>,
     ) -> anyhow::Result<()> {
-        let conn = if let Some(url) = config.get(CONFIG_REDIS_URL_KEY) {
+        let conn = if let Some(url) = config
+            .keys()
+            .find(|k| k.eq_ignore_ascii_case(CONFIG_REDIS_URL_KEY))
```
Comment (brooksmtownsend) [inline crates/provider-keyvalue-redis/src/lib.rs:306]: but but this is O(n) not O(1)! What if I have millions of keys!?!?!?

**Review by brooksmtownsend — APPROVED** [review]
Nice quality of life improvement 👍🏻

File: crates/provider-keyvalue-redis/src/lib.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/provider-keyvalue-redis/src/lib.rs:306]: I honestly did/am considering doing a check for both the original key and it lowercased... O(2) so much better than O(n), but we had this code "laying around" so to speak...

**Pull request merged**